sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveShim.scala

```diff
@@ -876,29 +876,21 @@ private[client] class Shim_v0_13 extends Shim_v0_12 {
       } else {
         logDebug(s"Hive metastore filter is '$filter'.")
         val tryDirectSqlConfVar = HiveConf.ConfVars.METASTORE_TRY_DIRECT_SQL
-        // We should get this config value from the metaStore. otherwise hit SPARK-18681.
-        // To be compatible with hive-0.12 and hive-0.13, In the future we can achieve this by:
-        // val tryDirectSql = hive.getMetaConf(tryDirectSqlConfVar.varname).toBoolean
-        val tryDirectSql = hive.getMSC.getConfigValue(tryDirectSqlConfVar.varname,
-          tryDirectSqlConfVar.defaultBoolVal.toString).toBoolean
         try {
           // Hive may throw an exception when calling this method in some circumstances, such as
           // when filtering on a non-string partition column when the hive config key
           // hive.metastore.try.direct.sql is false
```

Comment:
Update this comment?

Reply:
Hmm I think this comment is still accurate right? it explains why we need to catch the exception.

Reply:
> Yes, now if Spark is querying partitions from a remote HMS which throws exception even if hive.metastore.try.direct.sql is set to true, Spark will fallback ...

I think the intention of this PR is Hive also throws exception even the config is set to true?

Reply:
Yeah Hive might throw exception even when it is true, which is what we discovered here. The comment, I think, is on the `try ... catch` clause though, and it uses the case to explain why we need to catch the exception here. I can add a sentence such as "It could also throw exception even if `hive.metastore.try.direct.sql` is set to true in certain cases" if that helps to understand the code.